sw-precache.js

```diff
@@ -119,11 +120,13 @@ module.exports = function(params) {
   params.logger(util.format('Total precache size is about %s for %d resources.',
     formatBytesAsString(cumulativeSize), relativeUrls.length));
 
-  var templateBuffer = fs.readFileSync(params.templateFilePath);
-  return _.template(templateBuffer, {
+  var template = fs.readFileSync(params.templateFilePath, {encoding: 'utf8'});
+  var populatedTemplate = _.template(template, {
     handleFetch: params.handleFetch,
     importScripts: params.importScripts ? params.importScripts.map(JSON.stringify).join(',') : null,
     includeCachePolyfill: params.includeCachePolyfill,
     precacheConfig: JSON.stringify(precacheConfig)
   });
+
+  return resumer().queue(populatedTemplate).end();
```

Comment:
@addyosmani had suggested returning a stream. It might be that a stream was preferable to returning a string (which is what it was previously doing), and using callbacks is preferable to both of those.

One consideration is that this should be generic enough to produce output that could feed into either `gulp` or `grunt` (and potentially other node-based build tools out there). I'm assuming that, if properly written, there doesn't have to be anything `gulp`-specific about the callback.

@addyosmani, since streams were your suggestion, do you have any concerns about callbacks?